File: ros_learning_cpp/src/news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class NewsStation:public rclcpp::Node{
    public:
    NewsStation():Node("News_Station"){

        this->declare_parameter("message_to_publish", "Hi, new News");
        msg_data_ = this->get_parameter("message_to_publish").as_string();
        publisher_ = this->create_publisher<example_interfaces::msg::String>("news_station_pub",10);
        timer_ = this->create_wall_timer(std::chrono::milliseconds(500),
                std::bind(&NewsStation::publish_news,this));
        RCLCPP_INFO(this->get_logger(),"News Node started...");


    }
    private:
    void publish_news(){
        auto msg = example_interfaces::msg::String();
        msg.data = msg_data_;
        publisher_->publish(msg);
    }
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
    std::string msg_data_;
};

int main(int argc, char **argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NewsStation>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;

}

File: ros_learning_pkg/ros_learning_pkg/news_station.py
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String
class NewStation(Node):
    def __init__(self):
        super().__init__("news_station") 
        self.declare_parameter("message_to_publish", "hello") # <parameter name>, <default value>
        self.declare_parameter("publish_interval", 1.0)
        
        self.msgData_ = self.get_parameter("message_to_publish").value
        self.interval_ = self.get_parameter("publish_interval").value
        
        self.publiser_ = self.create_publisher(String, "news_station_pub",10 ) # msg datatype, topic name, queue length
        self.timer_ = self.create_timer(self.interval_,self.publish_news)
        self.get_logger().info("news_station started...")
        
        
    def publish_news(self):
        msg = String()
        msg.data = self.msgData_
        self.publiser_.publish(msg)



def main(args=None):
    rclpy.init(args=args)
    node = NewStation()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == 'main':
    main()